src/Screen/OpenGL/Init.cpp

```diff
@@ -322,12 +322,15 @@ OpenGL::SetupViewport(UnsignedPoint2D size)
   glViewport(0, 0, size.x, size.y);
 
 #ifdef SOFTWARE_ROTATE_DISPLAY
-  projection_matrix = glm::rotate(glm::mat4(1),
-                                  (GLfloat)Angle::Degrees(OrientationToRotation(display_orientation)).Radians(),
-                                  glm::vec3(0, 0, 1));
+  glm::mat4 rot_matrix = glm::rotate(
+    glm::mat4(1),
+    (GLfloat)Angle::Degrees(OrientationToRotation(display_orientation)).Radians(),
+    glm::vec3(0, 0, 1));
   OrientationSwap(size, display_orientation);
+#else
+  glm::mat4 rot_matrix = glm::mat4(1);  // no rotation
```

Comment:
This adds overhead for targets which have no software rotation, by multiplying this matrix which is a no-op. Can we avoid that overhead?